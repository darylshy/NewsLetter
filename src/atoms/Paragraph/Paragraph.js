import React from 'react';
import PropTypes from 'prop-types';
import classes from './Paragraph.module.scss';

const Paragraph = props => {
  return (
    <p className={classes.Paragraph}>
      {props.text}
      {props.children}
    </p>
  );
};

Paragraph.defaultProps = {
  text:
    'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Culpa incidunt possimus amet, est repellat sit architecto voluptatum magnam saepe, laborum maxime deleniti id corporis reiciendis odit nostrum! Alias, beatae dignissimos.'
};

Paragraph.propTypes = {
  text: PropTypes.string.isRequired
};

export default Paragraph;
