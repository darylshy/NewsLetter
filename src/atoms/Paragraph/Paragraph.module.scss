@import "../../styles/global/globals.scss";
.Paragraph {
  font-family: "Azo-Sans-Light";
  color: $font-grey;
  margin: 0;
  @include desktop {
    font-size: 1.2rem;
  }
  @include tabletL {
    font-size: 1.2rem;
  }
  @include tabletP {
    font-size: 1.2rem;
  }
  @include mobile {
    font-size: 1.2rem;
  }
}
