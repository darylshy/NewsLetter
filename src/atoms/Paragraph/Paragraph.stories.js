import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import React from 'react';

import Paragraph from './Paragraph';

storiesOf('Atoms/Paragraph', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add(
    'Standard Paragraph Element',
    () => <Paragraph style={{ color: '#94aebf' }} text={text('text')} />,
    {
      info: {
        text: 'Paragraph Element'
      }
    }
  );
