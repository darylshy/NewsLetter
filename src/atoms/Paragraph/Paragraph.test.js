import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Paragraph from './Paragraph';
import React from 'react';

configure({ adapter: new Adapter() });

describe('<Paragraph />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Paragraph text="some paragraph text" />);
  });
  it('should render an Paragraph component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should find a Paragraph', () => {
    expect(component.find('p')).toHaveLength(1);
  });
});
