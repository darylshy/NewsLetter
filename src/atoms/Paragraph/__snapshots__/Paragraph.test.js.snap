// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should render an Paragraph component 1`] = `ShallowWrapper {}`;
